docs/source/tutorial/train_loop.rst

```diff
@@ -0,0 +1,549 @@
+
+Chain
+.....
+
+- :class:`~chainer.Chain` is a class that can hold multiple links and/or functions. It is a subclass of :class:`~chainer.Link` and so it is also a :class:`~chainer.Link`.
+- This means that a :class:`~chainer.Chain` can contain parameters, which are the parameters of any links that it deeply contains.
+- In this way, :class:`~chainer.Chain` allows us to construct models with a potentially deep hierarchy of functions and links.
+- It is often convenient to use a single :class:`~chainer.Chain` that contains all of the layers (other chains, links, and functions) of the model. This is because we will need to optimize the model's parameters during training and if all of the parameters are contained by a single :class:`~chainer.Chain`, it turns out to be straightforward to pass these parameters into an optimizer (which we describe in more detail below).
+
+Variable
+........
+
+In Chainer, both the activations (that is, the inputs and outputs of functions
+and links) and the model parameters are instances of the
+:class:`~chainer.Variable` class. A :class:`~chainer.Variable` holds two
+arrays: a data array that contains the values that are read/written during the
+forward pass (or the parameter values), and a :attr:`~chainer.Variable.grad`
+array that contains the corresponding gradients that will be computed during
+the backward pass.
+
+A :class:`~chainer.Variable` can potentially contain two types of arrays as
+well, depending whether the array resides in CPU or GPU memory. By default,
+the CPU is used and these will be NumPy arrays. However, it is possible to move
+or create these arrays on the GPU as well, in which case they will be CuPy
+arrays. Fortunately, CuPy uses an API that is nearly identical to NumPy. This
+is convenient because in addition to making it easier for users to learn (there
+is almost nothing to learn if you are already familiar with NumPy), it often
+allows us to reuse the same code for both NumPy and CuPy arrays.
+
+Create our model as a subclass of Chain
+.......................................
+
+We can create our model be writing a new subclass of :class:`~chainer.Chain`.
+The two main steps are:
+
+1. Any :class:`~chainer.Link` objects (possibly also including other :class:`chainer.Chain` objects) that we wish to call during the forward computation of our :class:`~chainer.Chain` must first be supplied to the :class:`~chainer.Chain`'s :meth:`~chainer.Chain.__init__` method. After the :meth:`~chainer.Chain.__init__` method has been called, these :class:`~chainer.Link` objects will then be accessible as attributes of our :class:`~chainer.Chain` object. This means that we also need to provide the attribute name that we want to use for each :class:`~chainer.Link` object that is supplied. We do this by providing the attribute name and corresponding :class:`~chainer.Link` object as keyword arguments to :meth:`~chainer.Chain.__init__`, as we will do in the MLP chain below.
```

Comment:
This usage of `__init__` is not recommended now since v2. It is better to replace it by link registration based on `init_scope`.